test/clj/cider/nrepl/middleware/info_test.clj

```diff
@@ -163,6 +176,215 @@
   (is (re-find #"cider-nrepl" (:file (#'info/maybe-add-file {:ns (find-ns 'cider.nrepl.middleware.info)}))))
   (is (not (re-find #"/form-init[^/]*$" (:file (info/var-meta (eval '(do (in-ns 'cider.nrepl.middleware.info) (def pok 10)))))))))
 
+(deftest javadoc-info-unit-test
+  (testing "Get an HTTP URL for a sun/oracle Javadoc"
+    (let [reply (info/javadoc-info "java/lang/StringBuilder.html#capacity()")
+          url   (:javadoc reply)
+          jv    util/java-api-version
+          base  "http://docs.oracle.com/javase/%s/docs/api/java/lang/StringBuilder.html#capacity()"
+          exp   (format base jv)]
+      (is (= url exp))))
+
+  (testing "Get general URL for a clojure javadoc"
+    (let [reply    (info/javadoc-info "clojure/java/io.clj")
+          url      (:javadoc reply)
+          url-type (class url)
+          exp-type java.net.URL]
+      (is (= url-type exp-type))))
+
+  (testing "Get fall through URL type for other Javadocs (external libs?)"
+    (let [reply (info/javadoc-info "http://some/other/url")
+          url (:javadoc reply)]
+      (is (= url "http://some/other/url")))))
+
+;; TODO: These cljx need to be fleshed out. Right now just in to get 100% coverage
+;; TODO: ...or the cljx stuff could be removed since it's deprecated now?
```

Comment:
It's deprecated, but some people still use it, as they want to support older Clojure releases (older than 1.7).